pallets/circuit/src/lib.rs

```diff
@@ -1298,10 +1296,48 @@ impl<T: Config> Pallet<T> {
         current_weight
     }
 
+    /// At the XTX submission fn verify ensures that all of the gateways are active.
+    /// At either confirmation or revert attempt, ensure that all of the gateways are active, so that Executor won't be slashed.
+    pub fn ensure_all_gateways_are_active(xtx_id: XExecSignalId<T>) -> bool {
+        let xtx = match Machine::<T>::load_xtx(xtx_id) {
+            Ok(xtx) => xtx,
+            Err(_) => {
+                log::warn!("Failing to ensure_all_gateways_are_active. XTX not found.");
+                return false
+            },
+        };
+
+        for fsx in Machine::<T>::read_current_step_fsx(&xtx) {
+            let gateway_activity_overview = match <T as Config>::Xdns::read_last_activity(
+                fsx.input.target.clone(),
+            ) {
+                Some(gateway_activity_overview) => gateway_activity_overview,
+                None => {
+                    log::warn!("Failing to ensure_all_gateways_are_active. Target gateway is not registered in XDNS. Observe XDNS::gateway_activity_overview for more updates");
+                    return false
+                },
+            };
+
+            if gateway_activity_overview.is_active {
+                log::warn!(
+                    "Failing to ensure_all_gateways_are_active. Target gateway is currently not active. Observe XDNS::gateway_activity_overview for more updates"
+                );
+                return false
+            }
+        }
+        true
+    }
+
     pub fn process_revert_one(xtx_id: XExecSignalId<T>) -> Weight {
         const REVERT_WRITES: Weight = 2;
         const REVERT_READS: Weight = 1;
 
+        // Halt the Circuit reverting process if the gateway is not available for one of the FSX targets
+        if !Self::ensure_all_gateways_are_active(xtx_id) {
+            log::warn!("Failing to process_revert_one. Not all gateways are active.");
+            return 0
+        }
```

Comment:
Im not sure this weight number is correct, since you make a read for each gateway check